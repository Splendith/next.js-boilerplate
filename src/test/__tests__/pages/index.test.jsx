/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import ReduxWrapper from 'src/test/components/ReduxWrapper';
import Index from '~/pages/index';

describe('With Enzyme', () => {
  it('should shows "Hello world!"', () => {
    const app = shallow(<Index />);

    expect(
      app
        .find('Title')
        .children()
        .text(),
    ).toEqual('Hello world!');
  });
});

describe('With Snapshot Testing', () => {
  it('shallow snapshot', () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshot = shallowRenderer.render(<Index />);
    expect(snapshot).toMatchSnapshot();
  });
  it('full snapshot', () => {
    const component = renderer.create(
      <ReduxWrapper>
        <Index />
      </ReduxWrapper>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
