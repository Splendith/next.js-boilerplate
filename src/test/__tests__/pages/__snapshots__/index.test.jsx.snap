// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing full snapshot 1`] = `
<div
  className="Wrapper-sc-1v2z7vs-0 iWyySk"
>
  <div
    className="Box-sc-1v2z7vs-1 dBzgVg"
  >
    <h1
      className="Title-sc-1w6841p-0 fzvBpy"
    >
      Hello world!
    </h1>
    <div
      className="Image-sc-1jo6193-0 eWzmQJ"
    >
      <img
        alt="Smile"
        src="/static/img/smile.png"
      />
    </div>
    <p
      className="Welcome-sc-1jo6193-1 iFnFWv"
    >
      Welcome to this website!
    </p>
    <a
      href="/test/5?queryId=8"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      route=“/test/5?queryId=8“
    </a>
    <br />
    <a
      href="/test/7?queryId=9"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      route=“testurl“ params={{ id: 7, queryId: 9 }}
    </a>
  </div>
  <div
    className="Box-p23pin-0 eqtptK"
  >
    <h5>
      Redux global state
    </h5>
    <button
      aria-label={null}
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Add count
    </button>
     
    <button
      aria-label={null}
      className="btn btn-outline-secondary"
      onClick={[Function]}
      type="button"
    >
      Clear count
    </button>
    <br />
    <br />
    <div>
      Counter: 
      0
    </div>
    <div>
      Counter (persist): 
      0
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing shallow snapshot 1`] = `
<ForwardRef(LoadableComponent)>
  <Head>
    <title>
      Hello world!
    </title>
  </Head>
  <Title>
    Hello world!
  </Title>
  <Image>
    <img
      alt="Smile"
      src="/static/img/smile.png"
    />
  </Image>
  <Welcome>
    Welcome to this website!
  </Welcome>
  <LinkRoutes
    route="/test/5?queryId=8"
  >
    <a>
      route=“/test/5?queryId=8“
    </a>
  </LinkRoutes>
  <br />
  <LinkRoutes
    params={
      Object {
        "id": 7,
        "queryId": 9,
      }
    }
    route="testurl"
  >
    <a>
      route=“testurl“ params={{ id: 7, queryId: 9 }}
    </a>
  </LinkRoutes>
</ForwardRef(LoadableComponent)>
`;
